keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -423,8 +426,9 @@ def image_encoder(self):
         """
         if self._image_encoder is None:
             self._image_encoder = ImageEncoder(self.img_height, self.img_width)
-        if self.jit_compile:
+        if self.jit_compile and not self._image_encoder_comiled:
```

Comment:
Instead of adding a tracking variable for whether these are compiled, can we just nest this if statement inside the block that only runs if the encoder was none?

e.g. 
```
if self._image_encoder is None:
    self._image_encoder = ImageEncoder(self.img_height, self.img_width)
    if self.jit_compile:
        self._image_encoder.compile(jit_compile=True)
```

Reply:
Mhh.. do we have use cases where we need to explicitly control the compile phase?

Reply:
Not yet -- if we do, we can add guarding logic for that when it's necessary.

For now, compiling only at initialization time for the component models should bring us back to the functionality from pre-#989

Reply:
Ok

Reply:
Done